pkg/apiserver/authenticator/authn.go

```diff
@@ -96,12 +99,20 @@ func New(config AuthenticatorConfig) (authenticator.Request, error) {
 		authenticators = append(authenticators, serviceAccountAuth)
 	}
 
-	if len(config.KeystoneURL) > 0 {
-		keystoneAuth, err := newAuthenticatorFromKeystoneURL(config.KeystoneURL)
-		if err != nil {
-			return nil, err
+	if len(config.KeystoneConfig) > 0 {
```

Comment:
it'll try each in turn until one succeeds... if a bearer token is passed, the token auth would get called, if basic auth is passed, the password one would get called.